src/usr/share/batwatch/batwatch.py

```diff
@@ -53,124 +46,150 @@ def __eq__(self, other):
+        """Get status information about batteries connected to the device Batwatch is running
+            on, including the charge status, such that:
+            * If any battery is discharging, charge status is Discharging.
+              (Assume any status other than Charging or Fully Charged is equivalent to
+               Discharging.)
+            * Charge status is Fully Charged only if all batteries are Fully Charged.
+
+        returns: CompositeStatus describing the count and overall charge status of the
+            batteries.
         """
 
         device_names = self.upower_bus.EnumerateDevices()
-        charge_status = FULLY_CHARGED
 
         batteries = []
         for device_name in device_names:
             device = self.system_bus.get(UPOWER_BUS_NAME, device_name)
-            if device.PowerSupply is True and device.Type in (2, 3):  # Type 2 = Battery, Type 3 = UPS
+            # We only care about device types 2 and 3, which are batteries and UPS units,
+            #   respectively. The full list can be found here:
+            #   https://upower.freedesktop.org/docs/Device.html#Device:Type
+
+            # A device is considered a power supply if it powers the whole system.
+            if device.PowerSupply is True and device.Type in (2, 3):
                 batteries.append(device)
 
+        charge_status = FULLY_CHARGED
+
         if not batteries:
             charge_status = NO_BATTERY
 
-        for battery in batteries:
-            if battery.State == 1:
-                charge_status = CHARGING
-            elif battery.State == 4 and charge_status != CHARGING:
-                charge_status = FULLY_CHARGED
-            else:
-                charge_status = DISCHARGING
-                break
+        else:
+            for battery in batteries:
+                if battery.State == 1:
+                    charge_status = CHARGING
+                elif battery.State == 4 and charge_status != CHARGING:
+                    charge_status = FULLY_CHARGED
+                else:
+                    charge_status = DISCHARGING
+                    break
 
         return CompositeStatus(len(batteries), charge_status)
 
-    def _send_status_email(self, current_status, new_status):
-        """
-        Send an e-mail to the configured gpgmailer recipient when battery state changes.
-
-        current_status: The previous battery CompositeStatus.
-        new_status: The updated battery CompositeStatus.
-        """
-
-        email = gpgmailmessage.GpgMailMessage()
-        # get subject from self.config or else leave blank for gpgmailer default.
-        if (self.config['email_subject']):
-            email.set_subject(self.config['email_subject'])
-
-        body = self._build_email_body(current_status, new_status)
-        email.set_body(body)
+    def _build_email_body(self, prior_status, current_status):
+        """Creates an e-mail body with the battery status change information.
```

Comment:
Maybe say "message" instead of "body", cause, you know, "body" refers to the e-mail "body".

Reply:
My bad. This is named appropriately.